### PR TITLE
refactor: rewrite xontribs/jedi.xsh -> xontribs/jedi.py to take advan…

### PR DESCRIPTION
**Pull request by jnoortheen**
…tage of python tooling. 

Just fiddling with completers and saw this is the only core xontrib in `.xsh`. 

Probably `completers` need an API. Maybe a subclass of OrderedDict with methods to `add/remove`. What do you think about adding such API to completers? it will help testing and will close the gap with cli.

### CODE REVIEW

File: xontrib/jedi.py [hunk 1]
```diff
@@ -138,7 +138,6 @@ def create_completion(comp):
 xonsh.completers.base.complete_python = complete_jedi
 
 # Jedi ignores leading '@(' and friends
-completer remove python_mode
-
-completer add jedi_python complete_jedi '<python'
-completer remove python
+_aliases._remove_completer(["python_mode"])
+_aliases._add_one_completer("jedi_python", complete_jedi, "<python")
+_aliases._remove_completer(["python"])
```
Comment (scopatz) [inline xontrib/jedi.py:143]: Yeah, I agree. I don't love this.  There probably should be a public API. Maybe that would be best in another PR?

Comment (jnoortheen) [reply xontrib/jedi.py:143]: Ok. I will create a PR for that later. I will update the tests for this

**Comment by jnoortheen** [comment]
I have created an issue, to keep track of a Python API for the completers. Also the tests are fixed now @scopatz

**Comment by jnoortheen** [comment]
it also fixes failing elm-check pipeline that fails with miniconda@v1

File: tests/xontribs/test_jedi.py [hunk 2]
```diff
@@ -39,12 +39,8 @@ def jedi_xontrib(monkeypatch, source_path, jedi_mock, completer_mock):
...
 
 
-def test_completer_added(jedi_xontrib, completer_mock):
-    assert completer_mock.call_args_list == [
-        call(["remove", "python_mode"]),
-        call(["add", "jedi_python", "complete_jedi", "<python"]),
-        call(["remove", "python"]),
-    ]
+def test_completer_added(jedi_xontrib):
+    assert "xontrib.jedi" in sys.modules
```
Comment (daniel-shimon) [inline tests/xontribs/test_jedi.py:43]: This test is supposed to test the `jedi_python` completer was added and `python_mode` and `python` were removed.
Could you add code that still tests that? (e.g. by checking `builtins.__xonsh__.completers`)

**Review by daniel-shimon — CHANGES_REQUESTED** [review]

**Comment by daniel-shimon** [comment]
Hey @jnoortheen, could you rebase on master and remove 54e83c2f8dcb4536bc3c14ec3d350501c743ba55 (was superseded by #3984)

**Comment by jnoortheen** [comment]
> Hey @jnoortheen, could you rebase on master and remove [54e83c2](https://github.com/xonsh/xonsh/commit/54e83c2f8dcb4536bc3c14ec3d350501c743ba55) (was superseded by #3984)

Sure, will do

File: .github/workflows/elm.yml [hunk 3]
```diff
@@ -25,7 +25,7 @@ jobs:
         with:
           activate-environment: elm-xonsh-test
           environment-file: ci/environment-elm.yml
-          update-conda: true
+          auto-update-conda: true
```
Comment (daniel-shimon) [inline .github/workflows/elm.yml:28]: @scopatz is this ok/needed? I don't know too much about conda

Other than that I think this PR is ready to merge 🔥

Comment (jnoortheen) [reply .github/workflows/elm.yml:28]: without it would show warning. That miniconda action has renamed this variable long ago

**Review by daniel-shimon — APPROVED** [review]
Great! @scopatz ready to merge

**Review by gforsyth — APPROVED** [review]
Thanks for putting this in @jnoortheen and thanks @daniel-shimon for the extra review!
